code/modules/projectiles/guns/energy/stun.dm

```diff
@@ -30,7 +31,49 @@
 	origin_tech = list(TECH_COMBAT = 3, TECH_MATERIAL = 3, TECH_POWER = 2)
 	projectile_type = /obj/item/projectile/energy/electrode
 	max_shots = 8
+	var/emagged = FALSE
 
+/obj/item/gun/energy/stunrevolver/verb/rename_gun()
+	set name = "Name Gun"
+	set category = "Object"
+	set desc = "Click to rename your gun. If you're the detective."
+
+	var/mob/M = usr
+	if(!M.mind)
+		return FALSE
```

Comment:
Would need to check if mob is detective, no? That, or update the desc.

Reply:
I wouldn't restrict the renaming to just detectives.
(Also imho the renaming could just be removed at this point)

Reply:
It's a stylistic flair that I'd rather keep. Being stunned by The Gnomer is something everyone should experience.